src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int GROWTH_FACTOR = 2;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = hash(key) % table.length;
+        if (table[index] == null) {
+            table[index] = new Node<>(hash(key), key, value, null);
+            size++;
+        } else if (findNodeByKey(key, index) == null) {
+            Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+            Node<K, V> lastNode = getLastNode(index);
+            lastNode.next = newNode;
+            size++;
+        } else {
+            Node<K, V> currentNode = findNodeByKey(key, index);
+            currentNode.value = value;
+        }
+
+        if (size > (table.length * LOAD_FACTOR)) {
+            resize();
+        }
 
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key) % table.length;
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V> findNodeByKey(K key, int index) {
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode;
+            }
```

Comment:
![image](https://github.com/mate-academy/jv-hash-map/assets/156338484/8d774f25-b2c5-4d04-8e7b-9e43ffb53a67)
I see a duplication of code, try to get into separate method